### PR TITLE
Project: Refactor About pages

### PR DESCRIPTION
**Pull request by eatyourgreens**
Refactor project data-fetching to speed up project pages.
- remove `pages` from the default project request.
- add a `fetchProjectPages` helper which fetches the list of project page titles (for the About page navigation.)
- add `fetchProjectPages` to each page under `[owner]/[project]/about`.

## Package
app-project
lib-panoptes-js

## Linked Issue and/or Talk Post
Project data-fetching uses a relatively slow API query that runs on every project page (as part of `getDefaultPageProps`.) Here, 554ms, of 791ms total, is spent on the `/api/projects` request to build the PH-TESS home page.

Timings, from New Relic, for a single slow build of the PH-TESS home page. 554ms is spent running the query to /api/projects for the project data.

This PR speeds that query up by removing project pages from the default project data, instead requesting them only on the pages that use them.

## How to Review
The easiest way to test this might be to check the About pages for a real project:
https://local.zooniverse.org:3000/projects/msalmon/hms-nhs-the-nautical-health-service/about/team?env=production

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/app-project/src/helpers/fetchProjectPages/fetchProjectPages.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export default async function fetchProjectPages(project, env) {
+  try {
+    const { headers, host } = getServerSideAPIHost(env)
+    // the endpoint is /project/{project.id}/pages but the returned array is project_pages.
+    const response = await panoptes.get(`/projects/${project.id}/pages`, { env }, { ...headers }, host)
+    const projectPages = response?.body?.project_pages || []
+    /* Only the page titles and URL keys are needed
+    to build the navigation menu */
+    return projectPages.map(page =>{
```
Comment (goplayoutside3) [inline packages/app-project/src/helpers/fetchProjectPages/fetchProjectPages.js:13]:
```suggestion
    return projectPages.map(page => {
```

**Review by goplayoutside3 — COMMENTED** [review]
This looks like it works to me! 

1. I think it would be clearer if `fetchProjectPages` (plural) could be named something like `fetchProjectPagesTitles` to differentiate from the singular helper `fetchProjectPage`.

2. I noticed this new helper function doesn't have a spec file, and your recently added helper `fetchTeam` doesn't either. If a spec file isn't valuable in their use cases, I suggest adding a Readme to their directories noting their purpose and where they're used.

**Comment by eatyourgreens** [comment]
Good point about the tests. I’ll open another PR to add those.

**Review by goplayoutside3 — APPROVED** [review]
Looks good!

**Pull request merged**